net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,545 @@
+    # https://github.com/apple/ccs-calendarserver/blob/master/conf/caldavd-test-s2s.plist
+    ln -s ${prefix} \
+        ${destroot}${calendarserverdir}/Library/CalendarServer/memcached/_root
+}
+
+# Network configuration
+# hard-coded examples
+set host                host
+set domain              domain
+set tld                 tld
+set fullhost            ${host}.${domain}.${tld}
+set domaintld           ${domain}.${tld}
+set tls_certificate_name \
+                        ${fullhost}
+
+startupitem.create     yes
+
+startupitems \
+    name               calendarserver \
+    executable \
+        ${calendarserverdir}/Library/CalendarServer/${ccsname}/bin/caldavd
+
+# use defaults, edit launch daemon plists in post-activate below
+# executable ${prefix}/sbin/nginx -g "daemon off;" -p ${calendarserverdir}/Library/CalendarServer/logs -c ${calendarserverdir}/Library/CalendarServer/etc/nginx.conf
+# pidfile auto ${calendarserverdir}/Library/CalendarServer/logs/nginx.pid
+startupitems-append \
+    name               calendarserver_proxy \
+    executable         ${prefix}/sbin/nginx \
+    pidfile            auto
+
+post-activate {
+    # personal data outside of ${prefix}
+    if { ![file exists \
+        /private/var/calendarserver/Library/CalendarServer] } {
+        xinstall -m 0755 -o ${calendarserverUser} -g _calendar -d \
+            /private/var/calendarserver/Library/CalendarServer
+    }
+    if { ![file exists \
+        /private/var/calendarserver/Library/CalendarServer/Data] } {
+        xinstall -m 0700 -o ${calendarserverUser} -g _calendar -d \
+            /private/var/calendarserver/Library/CalendarServer/Data
+    }
+
+    # package/install calendarserver binaries
+    system -W ${worksrcpath} \
+        "sh <<PACKAGE_CALENDARSERVER
+            # Don't use SecureTransport, prefer MacPorts Python binaries and libraries
+            export USE_OPENSSL=1
+            export PATH='${workpath}/bin:${prefix}/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/bin:$env(PATH)'
+            export C_INCLUDE_PATH='${prefix}/include:/usr/include:/usr/local/include'
+            export LD_LIBRARY_PATH='${prefix}/lib:${prefix}/lib/openssl-1.0:/usr/lib:/usr/local/lib'
+            export CPPFLAGS='-I${prefix}/include -I/usr/include -I/usr/local/include'
+            export LDFLAGS='-L${prefix}/lib -L${prefix}/lib/openssl-1.0-L/usr/lib -L/usr/local/lib'
+            export DYLD_LIBRARY_PATH='${prefix}/lib:${prefix}/lib/openssl-1.0:/usr/lib:/usr/local/lib'
+            ./bin/package \
+                ${calendarserverdir}/Library/CalendarServer/${ccsname}
+PACKAGE_CALENDARSERVER
+"
+
+    # use network settings for installed example configuration
+    set fullhost [exec /bin/hostname -f]
+    set host [lindex [split ${fullhost} .] 0]
+    set domaintld [join [lrange [split ${fullhost} .] end-1 end] .]
+    set domain [lindex [split ${domaintld} .] 0]
+    set tld [lindex [split ${domaintld} .] end]
```

Comment:
It strikes me as strange to want to separate the hostname ("fullhost") into these parts. If my hostname is "box.example.com", what is the significance to calendarserver that the "host" is "box", the "domain" is "example" and the "tld" is "com"? Why does it care about anything other than the entire hostname?

The code isn't parsing the hostname correctly unless it has exactly two periods in it, and I'm not sure that there is a way to parse it correctly, given the existence of ccTLDs, unless you keep a list of all known ccTLDs.

If the hostname is "box.internal.example.co.uk", it sets "host" to "box", "domain" to "co" and "tld" to "uk", ignoring the "internal.example" part, and it will write "box.co.uk" to the config file.

What if the hostname is just "example.com"? "host" and "domain" both get set to "example" and "tld" gets set to "com", and it writes "example.example.com" to the config file.

Reply:
Thanks. I modified the logic to handle this more robustly.

I used to have this really fancy, inscrutable perl command that handled all this, but I replaced it with a KISS approach that handles the vast majority of situations.

Also, this just provides an **example** configuration. Users must necessarily check and edit all config files for their own deployments. There's simply no way for the Portfile to automatically generate config files appropriate for deployment in all situations, so the fine details of these example config files are not that important.